src/v/security/audit/audit_log_manager.cc

```diff
@@ -522,8 +522,10 @@ ss::future<> audit_client::produce(
     // here should usually be 1-2, since the default per-shard queue limit
     // is 1MiB, which is also the default for kafka_batch_max_bytes.
     // TODO(oren): a configurabale ratio might be better
-    [[maybe_unused]] auto max_concurrency
-      = _max_buffer_size / config::shard_local_cfg().kafka_batch_max_bytes();
+    auto max_concurrency = std::clamp<size_t>(
```

Comment:
STupid question: why not `std::max<size_t>(1, _max_buffer_size / config::shard_local_cfg().kafka_batch_max_bytes())`?

Reply:
That would work too. I've added the upper bound because we don't need more parallelism than the number of records, but that part is not strictly necessary.

Reply:
fair enough, approved